jinja2/filters.py

```diff
@@ -518,9 +518,17 @@ def do_int(value, default=0, base=10):
     can also override the default base (10) in the second
     parameter, which handles input with prefixes such as
     0b, 0o and 0x for bases 2, 8 and 16 respectively.
+    The base is ignored for non-string values.
     """
+    # only string types support int() with explicit base
+    if isinstance(value, string_types) and base != 10:
+        try:
+            return int(value, base)
+        except ValueError:
+            return default
```

Comment:
I realized this, and considered it a bug as well. Treating strings with representing a floating point number as decimal, when hexadecimal was specified, while respecting the base when the string represents an integer, is certainly wrong.

You could either say that floating point semantics aren't specified for non-decimal numbers. Then my code would behave correctly and as expected by returning the default.

Or one could insist that floating points should be supported consistently for all bases. This could be implemented like below. Let me know what you think.

```python
try:
        if isinstance(value, string_types):
            return [int(x, base) for x in value.split('.', 1)][0]
        else:
            try:
                return int(value)
            except (TypeError, ValueError):
                return int(float(value))
    except (TypeError, ValueError):
        return default
```